server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala

```diff
@@ -103,7 +103,7 @@ class ResourceServiceBuilder[F[_]] private (
               }
               .flatMap { path =>
                 StaticFile.fromResource(
-                  path.toString,
+                  path.toString.replaceAll("\\\\", "/"),
```

Comment:
I wonder if it makes sense to be more OS-specific here.
E.g. on JVM there's a `java.io.File.separator` constant which can be used here for the sake of the replacement.

Reply:
As I understand it is that we need to provide Linux separator regardless if the code runs on Windows or Linux. Do you have a suggestion on what to change?

Reply:
Right, Sergey's proposal is not to use `java.io.File.separator` for the _replacement_, but for the pattern to replace i.e. the `"\\\\"`.

Reply:
Yes, correct. Sorry if I made it unclear. Basically, if we assume that path segment separator is always a single char regardless of the target OS, then we can make use of
```java
public String replace(char oldChar, char newChar)
```
from `String`:
```suggestion
                  path.toString.replace(File.separatorChar, '/'),
```
I guess it should be faster that `replaceAll` (which takes a regex for the first parameter).
Also, `replace` does nothing (no-op) if `oldChar` and `newChar` are the same.

Reply:
Oh, great that it's `separatorChar`! That's very nice :)

Reply:
Very nice. I like your solution better. I will try this and test on Windows and Linux if it works just to be safe.

Reply:
I have tested it and it works both on Windows and Linux.